### ISSUE TITLE
ngx_cache_purge build errors with current nginx head

### ISSUE THREAD

**Issue by jlporter** [FRiCKLE/ngx_cache_purge#26]
Building with the most recent nginx from trunk as of today (http://hg.nginx.org/nginx/archive/tip.zip), I'm receiving a number of build errors with this module.

```
nginx/ngx_cache_purge/ngx_cache_purge_module.c: In function ‘ngx_http_fastcgi_cache_purge_conf’:
nginx/ngx_cache_purge/ngx_cache_purge_module.c:274:30: warning: comparison between pointer and integer [enabled by default]
     if (flcf->upstream.cache != NGX_CONF_UNSET_PTR
                              ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c:275:33: warning: comparison between pointer and integer [enabled by default]
         && flcf->upstream.cache != NULL)
                                 ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c:291:26: warning: assignment makes integer from pointer without a cast [enabled by default]
     flcf->upstream.cache = ngx_shared_memory_add(cf, &value[1], 0,
                          ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c:293:30: warning: comparison between pointer and integer [enabled by default]
     if (flcf->upstream.cache == NULL) {
                              ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c: In function ‘ngx_http_fastcgi_cache_purge_handler’:
nginx/ngx_cache_purge/ngx_cache_purge_module.c:324:58: error: invalid type argument of ‘->’ (have ‘int’)
     if (ngx_http_cache_purge_init(r, flcf->upstream.cache->data,
                                                          ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c: In function ‘ngx_http_proxy_cache_purge_conf’:
nginx/ngx_cache_purge/ngx_cache_purge_module.c:463:30: warning: comparison between pointer and integer [enabled by default]
     if (plcf->upstream.cache != NGX_CONF_UNSET_PTR
                              ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c:464:33: warning: comparison between pointer and integer [enabled by default]
         && plcf->upstream.cache != NULL)
                                 ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c:480:26: warning: assignment makes integer from pointer without a cast [enabled by default]
     plcf->upstream.cache = ngx_shared_memory_add(cf, &value[1], 0,
                          ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c:482:30: warning: comparison between pointer and integer [enabled by default]
     if (plcf->upstream.cache == NULL) {
                              ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c: In function ‘ngx_http_proxy_cache_purge_handler’:
nginx/ngx_cache_purge/ngx_cache_purge_module.c:513:58: error: invalid type argument of ‘->’ (have ‘int’)
     if (ngx_http_cache_purge_init(r, plcf->upstream.cache->data,
                                                          ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c: In function ‘ngx_http_scgi_cache_purge_conf’:
nginx/ngx_cache_purge/ngx_cache_purge_module.c:594:30: warning: comparison between pointer and integer [enabled by default]
     if (slcf->upstream.cache != NGX_CONF_UNSET_PTR
                              ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c:595:33: warning: comparison between pointer and integer [enabled by default]
         && slcf->upstream.cache != NULL)
                                 ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c:611:26: warning: assignment makes integer from pointer without a cast [enabled by default]
     slcf->upstream.cache = ngx_shared_memory_add(cf, &value[1], 0,
                          ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c:613:30: warning: comparison between pointer and integer [enabled by default]
     if (slcf->upstream.cache == NULL) {
                              ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c: In function ‘ngx_http_scgi_cache_purge_handler’:
nginx/ngx_cache_purge/ngx_cache_purge_module.c:644:58: error: invalid type argument of ‘->’ (have ‘int’)
     if (ngx_http_cache_purge_init(r, slcf->upstream.cache->data,
                                                          ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c: In function ‘ngx_http_uwsgi_cache_purge_conf’:
nginx/ngx_cache_purge/ngx_cache_purge_module.c:748:30: warning: comparison between pointer and integer [enabled by default]
     if (ulcf->upstream.cache != NGX_CONF_UNSET_PTR
                              ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c:749:33: warning: comparison between pointer and integer [enabled by default]
         && ulcf->upstream.cache != NULL)
                                 ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c:765:26: warning: assignment makes integer from pointer without a cast [enabled by default]
     ulcf->upstream.cache = ngx_shared_memory_add(cf, &value[1], 0,
                          ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c:767:30: warning: comparison between pointer and integer [enabled by default]
     if (ulcf->upstream.cache == NULL) {
                              ^
nginx/ngx_cache_purge/ngx_cache_purge_module.c: In function ‘ngx_http_uwsgi_cache_purge_handler’:
nginx/ngx_cache_purge/ngx_cache_purge_module.c:798:58: error: invalid type argument of ‘->’ (have ‘int’)
     if (ngx_http_cache_purge_init(r, ulcf->upstream.cache->data,
                                                          ^
```

**Comment by PiotrSikora** [comment]
I'm aware of this, but because of the version checks, the fix won't be pushed before official 1.7.9 release (which should be released tomorrow).

**Comment by PiotrSikora** [comment]
Fixed in 065f0cae56fc5720db636885ce5b2fbaf6f09f8c, released in 2.3.
